### PR TITLE
PHRAS-595 - Add password reset route

### PR DESCRIPTION
**Pull request by aztech-dev**

### CODE REVIEW

File: lib/Alchemy/Phrasea/Authentication/Exception/RecoveryException.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Alchemy\Phrasea\Authentication\Exception;
+
+class RecoveryException extends \RuntimeException
+{
+
+}
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Authentication/Exception/RecoveryException.php]: Missing EOL

File: lib/Alchemy/Phrasea/Authentication/RecoveryService.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        /** @var MailRequestPasswordUpdate $mail */
+        $mail = MailRequestPasswordUpdate::create($this->app, $receiver);
+        $mail->setLogin($user->get_login());
+        $mail->setButtonUrl($url);
+        $mail->setExpiration($expirationDate);
+
+        $this->mailer->deliver($mail);
+        $this->app->addFlash('info', _('phraseanet:: Un email vient de vous etre envoye'));
+    }
+}
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Authentication/RecoveryService.php]: EOF

File: lib/Alchemy/Phrasea/Authentication/RecoveryService.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            throw new InvalidArgumentException('Invalid email address', 0, $e);
+        }
+
+        $url = $this->urlGenerator->generate(
+            'login_renew_password',
+            array('token' => $token),
+            UrlGeneratorInterface::ABSOLUTE_URL
+        );
+
+        /** @var MailRequestPasswordUpdate $mail */
```
Comment (bburnichon) [inline lib/Alchemy/Phrasea/Authentication/RecoveryService.php]: useless phpdoc, create should already have typehint.

File: lib/classes/User/Adapter.php [hunk 4]
```diff
@@ -355,16 +355,18 @@ public static function unsetInstance($id)
...
-     * @param  type         $id
-     * @param  Application  $app
+     * @param  int $id
+     * @param  Application $app
      * @return User_Adapter
+     * @throws Exception
      */
     public static function getInstance($id, Application $app)
     {
         if (is_int((int) $id) && (int) $id > 0) {
```
Comment (bburnichon) [inline lib/classes/User/Adapter.php]: Is that a check of int conversion?

**Comment by aztech-dev** [comment]
@bburnichon Ready for merge once you review.

**Comment by bburnichon** [comment]
:+1:

**Pull request merged**